content/client-lib-development-guide/features.textile

```diff
@@ -1473,6 +1586,85 @@ enum StatsIntervalGranularity:
   DAY
   MONTH
 
+class DeviceDetails:
+  id: String
+  platform: DevicePlatform
+  formFactor: DeviceFormFactor
+  clientId: String?
+  metadata: JsonObject
+  updateToken: String
+  push: DevicePushDetails
+
+class DevicePushDetails:
+  transportType: DevicePushTransportType
+  transportId: String
+  transportToken: String?
+  state: .Active | .Failing | .Failed
+  errorReason: ErrorInfo?
+  metadata: JsonObject
+
+class LocalDevice extends DeviceDetails:
+  resetId()
+  resetUpdateToken() => io
+
+class Push:
+  publish(recipient: JsonObject, data: JsonObject) => io
+  admin: PushAdmin
+
+  // Only on platforms that support receiving notifications:
+
+  activate(
+    registerCallback: ((ErrorInfo?, DeviceDetails?) -> io String)?,
+    // Only on platforms that, after first set, can update later its push
+    // device details:
+    updateFailedCallback: ((ErrorInfo) ->)
+  ) => io ErrorInfo? // RSH3
+  deactivate(
+    deregisterCallback: ((ErrorInfo?, deviceId: String?) -> io)?
+  ) => io ErrorInfo?
+
+class PushAdmin:
+  deviceRegistrations: PushDeviceRegistrations
+  channelSubscriptions: PushChannelSubscriptions
+
+class JsonObject:
+  // Platform-dependant, typically a Dict-like object
+
+class PushDeviceRegistrations:
+  save(DeviceDetails) => io
+  get(params: Dict<String, String>?) => io PaginatedResult<DeviceDetails>
+  remove(params: Dict<String, String>?) => io
+
+class PushChannelSubscriptions:
+  save(PushChannelSubscription) => io
+  listChannels() => io PaginatedResult<String>
+  get(params: Dict<String, String>?) => io PaginatedResult<PushChannelSubscription>
+  remove(params: Dict<String, String>?) => io
+
+enum DevicePushTransportType:
+  "fcm" // PTT1
+  "gcm" // PTT1
+  "apns" // PTT1
+  "web" // PTT1
+
+enum DevicePlatform:
+  "android" // PPT1
+  "ios" // PPT1
+  "browser" // PPT1
+
+enum DeviceFormFactor:
+  "mobile" // PDT1
+  "tablet" // PDT1
+  "desktop" // PDT1
+  "embedded" // PDT1
```

Comment:
I can detect many device forms in the Cocoa ecosystem:

 - Phone
 - Tablet
 - Watch
 - TV
 - Car
 - Desktop
 - Simulator

Should we update this or it's fine for now?